File: ServiTec/ServiTec/Domain/Models/Usuari.cs
// ============================================================================
// Projecte:      ServiTec - Sistema de Gestió de Restaurants (TFG)
// Autor:         Luca Klein
// Titulació:     Grau en Enginyeria Informàtica (4t Curs)
// Institució:    Universitat de Girona (UdG)
// Fitxer:        Usuari.cs
// Descripció:    Entitat de domini que representa els usuaris del sistema 
//                (cambrers, personal de cuina i administradors), gestionant 
//                les seves credencials, rols i permisos d'accés.
// ============================================================================

using System;
using System.Collections.Generic;

namespace ServiTec.Domain.Models;

/// <summary>
/// Entitat que representa un usuari o treballador del sistema ServiTec.
/// </summary>
public partial class Usuari
{
    /// <summary>
    /// Identificador únic de l'usuari.
    /// </summary>
    public int IdUsuari { get; set; }

    /// <summary>
    /// Nom d'usuari utilitzat per a l'autenticació al sistema.
    /// </summary>
    public string NomUsuari { get; set; } = null!;

    /// <summary>
    /// Contrasenya de l'usuari encriptada/hashing.
    /// </summary>
    public string Contrasenya { get; set; } = null!;

    /// <summary>
    /// Indica si l'usuari està actiu i pot accedir a l'aplicació.
    /// </summary>
    public bool Actiu { get; set; }

    /// <summary>
    /// Indica si l'usuari disposa de permisos d'administrador.
    /// </summary>
    public bool Admin { get; set; }

    /// <summary>
    /// Rol assignat a l'usuari dins del sistema (ex. "Cambrer", "Cuina", "Admin").
    /// </summary>
    public string Rol { get; set; } = null!;

    /// <summary>
    /// Col·lecció de comandes creades directament per aquest usuari.
    /// </summary>
    public virtual ICollection<Comanda> Comanda { get; set; } = new List<Comanda>();

    /// <summary>
    /// Col·lecció d'associacions a comandes en què l'usuari ha intervingut.
    /// </summary>
    public virtual ICollection<LiniaUsuari> LiniaUsuaris { get; set; } = new List<LiniaUsuari>();
}

File: ServiTec_frontend/app/src/main/java/com/example/servitec_frontend/data/model/Usuari.kt
// ============================================================================
// Projecte:      ServiTec - Sistema de Gestió de Restaurants (TFG)
// Autor:         Luca Klein
// Titulació:     Grau en Enginyeria Informàtica (4t Curs)
// Institució:    Universitat de Girona (UdG)
// Fitxer:        UsuariDTO.kt
// Descripció:    Objectes de transferència de dades (DTOs) per a la gestió,
//                creació i actualització d'usuaris del sistema.
// ============================================================================

package com.example.servitec_frontend.data.model

/**
 * DTO que representa la informació pública d'un usuari registrat al sistema,
 * incloent-hi el seu rol, estat d'activitat i el token JWT d'accés en cas d'autenticació.
 *
 * @property idUsuari Identificador únic de l'usuari a la base de dades.
 * @property nomUsuari Nom d'usuari utilitzat per accedir a la plataforma.
 * @property rol Rol o perfil assignat (p. ex., Cambrer, Cuiner, Administrador).
 * @property admin Indica si l'usuari disposa de privilegis d'administració.
 * @property actiu Estat del compte d'usuari (actiu o desactivat).
 * @property token Token de sessió JWT proporcionat pel backend.
 */
data class UsuariDTO(
    val idUsuari: Int,
    val nomUsuari: String,
    val rol: String,
    val admin: Boolean,
    val actiu: Boolean,
    val token: String? = null
)

/**
 * DTO utilitzat per enviar les dades d'actualització d'un usuari existent.
 *
 * @property nomUsuari Nou nom d'usuari assignat.
 * @property contrasenya Nova contrasenya o la mateixa reescrita.
 * @property actiu Nou estat d'activació del compte.
 * @property admin Permisos d'administració assignats.
 * @property rol Rol funcional actualitzat.
 */
data class UpdateUsuariDTO(
    val nomUsuari: String,
    val contrasenya: String,
    val actiu: Boolean,
    val admin: Boolean,
    val rol: String
)

/**
 * DTO utilitzat per a la creació i registre d'un nou usuari al sistema.
 *
 * @property nomUsuari Nom d'usuari requerit per a l'accés.
 * @property contrasenya Contrasenya d'accés inicial.
 * @property actiu Estat inicial del compte (per defecte actiu).
 * @property admin Defineix si el nou usuari tindrà rols administratius.
 * @property rol Rol o funció assignada dins del restaurant.
 */
data class CreateUsuariDTO(
    val nomUsuari: String,
    val contrasenya: String,
    val actiu: Boolean,
    val admin: Boolean,
    val rol: String
)